@ykwd hi~ I looked into the latest CI run.

The remaining failure appears to be a runner interruption in `build-wheel-cu13`, not a code/build error from this PR. The regular build jobs, build-flags jobs, Docker image build, and wheel tests all passed. The failed cu13 job ended with:

```text
ninja: build stopped: interrupted by user.
The runner has received a shutdown signal.
Process completed with exit code 143.
```
I think rerunning the failed job should be enough here.

--- a/mooncake-store/include/utils.h
+++ b/mooncake-store/include/utils.h
@@ -207,6 +207,9 @@ std::string expected_to_str(const tl::expected<T, ErrorCode>& expected) {
     } catch (const std::exception&) {
         return std::nullopt;  // Failed to parse number
     }
+    if (value < 0) {
+        return std::nullopt;
+    }
 
     if (pos >= s.length()) {
         // No unit specified, assume bytes

--- a/mooncake-store/src/real_client.cpp
+++ b/mooncake-store/src/real_client.cpp
@@ -932,17 +932,6 @@ inline std::string get_config(const ConfigDict &config, const std::string &key,
     return (it != config.end()) ? it->second : default_value;
 }
 
-inline std::string trim_config_value(std::string value) {
-    value.erase(0, value.find_first_not_of(" \t\r\n"));
-    auto last = value.find_last_not_of(" \t\r\n");
-    if (last == std::string::npos) {
-        value.clear();
-    } else {
-        value.erase(last + 1);
-    }
-    return value;
-}
-
 inline std::optional<size_t> get_config_size(const ConfigDict &config,
                                              const std::string &key,
                                              size_t default_value) {
@@ -950,26 +939,14 @@ inline std::optional<size_t> get_config_size(const ConfigDict &config,
     if (it == config.end()) {
         return default_value;
     }
-    const std::string value = trim_config_value(it->second);
-    if (value.empty() || value[0] == '-') {
-        LOG(ERROR) << "Invalid value for config key '" << key
-                   << "': " << it->second;
-        return std::nullopt;
-    }
 
-    auto parsed_size_opt = try_string_to_byte_size(value);
+    auto parsed_size_opt = try_string_to_byte_size(it->second);
     if (!parsed_size_opt.has_value()) {
         LOG(ERROR) << "Invalid size value for config key '" << key
                    << "': " << it->second;
         return std::nullopt;
     }
-    uint64_t parsed_size = parsed_size_opt.value();
-    if (parsed_size > std::numeric_limits<size_t>::max()) {
-        LOG(ERROR) << "Value out of range for config key '" << key
-                   << "': " << it->second;
-        return std::nullopt;
-    }
-    return static_cast<size_t>(parsed_size);
+    return static_cast<size_t>(parsed_size_opt.value());
 }
 }  // namespace
 
@@ -1011,19 +988,19 @@ tl::expected<void, ErrorCode> RealClient::setup_internal(
     std::string ipc_socket_path =
         get_config(config, CONFIG_KEY_IPC_SOCKET_PATH);
 
-    // Validate size parameters are within acceptable ranges
-    if ((global_segment_size != 0 && global_segment_size < MIN_SEGMENT_SIZE) ||
-        global_segment_size > MAX_SEGMENT_SIZE) {
-        LOG(ERROR) << "Invalid " << CONFIG_KEY_GLOBAL_SEGMENT_SIZE << ": "
-                   << global_segment_size << ", must be 0 or between "
-                   << MIN_SEGMENT_SIZE << " and " << MAX_SEGMENT_SIZE;
-        return tl::unexpected(ErrorCode::INVALID_PARAMS);
-    }
-    if ((local_buffer_size != 0 && local_buffer_size < MIN_SEGMENT_SIZE) ||
-        local_buffer_size > MAX_SEGMENT_SIZE) {
-        LOG(ERROR) << "Invalid " << CONFIG_KEY_LOCAL_BUFFER_SIZE << ": "
-                   << local_buffer_size << ", must be 0 or between "
-                   << MIN_SEGMENT_SIZE << " and " << MAX_SEGMENT_SIZE;
+    // A size of 0 keeps the pure client/server setup semantics.
+    auto validate_size = [](const char *key, size_t value) {
+        if ((value != 0 && value < MIN_SEGMENT_SIZE) ||
+            value > MAX_SEGMENT_SIZE) {
+            LOG(ERROR) << "Invalid " << key << ": " << value
+                       << ", must be 0 or between " << MIN_SEGMENT_SIZE
+                       << " and " << MAX_SEGMENT_SIZE;
+            return false;
+        }
+        return true;
+    };
+    if (!validate_size(CONFIG_KEY_GLOBAL_SEGMENT_SIZE, global_segment_size) ||
+        !validate_size(CONFIG_KEY_LOCAL_BUFFER_SIZE, local_buffer_size)) {
         return tl::unexpected(ErrorCode::INVALID_PARAMS);
     }
 

--- a/mooncake-store/tests/pybind_client_test.cpp
+++ b/mooncake-store/tests/pybind_client_test.cpp
@@ -973,6 +973,7 @@ TEST_F(RealClientTest, ErrSetupWithInvalidConfigDictSize) {
     const InvalidSizeCase invalid_size_cases[] = {
         {"localhost:17816", "50%", "16MB"},
         {"localhost:17817", "16MB", "16XB"},
+        {"localhost:17818", "-5", "16MB"},
     };
 
     for (const auto& test_case : invalid_size_cases) {

--- a/mooncake-store/tests/utils_test.cpp
+++ b/mooncake-store/tests/utils_test.cpp
@@ -23,6 +23,20 @@ TEST(UtilsTest, ByteSizeToString) {
     EXPECT_EQ(byte_size_to_string(15 * 1024 * 1024 + 44048), "15.04 MB");
 }
 
+TEST(UtilsTest, StringToByteSize) {
+    auto parsed = try_string_to_byte_size("16 MB");
+    ASSERT_TRUE(parsed.has_value());
+    EXPECT_EQ(parsed.value(), 16ULL * 1024 * 1024);
+
+    parsed = try_string_to_byte_size("0");
+    ASSERT_TRUE(parsed.has_value());
+    EXPECT_EQ(parsed.value(), 0);
+
+    EXPECT_FALSE(try_string_to_byte_size("-5").has_value());
+    EXPECT_FALSE(try_string_to_byte_size("16XB").has_value());
+    EXPECT_EQ(string_to_byte_size("-5"), 0);
+}
+
 TEST(UtilsTest, StringToBool) {
     EXPECT_EQ(string_to_bool("1"), true);
     EXPECT_EQ(string_to_bool("true"), true);

